src/bci_build/package/versions.py

```diff
@@ -155,7 +155,7 @@ def format_version(ver: str, format: ParseVersion) -> str:
     1.2.0
 
     """
-    v = version.parse(ver)
+    v = version.parse(ver.replace(":", "!").partition("~")[0])
```

Comment:
Can you please split this into a separate commit explaining why?

Reply:
done.